src/app/cras/edit-cra/edit-cra.component.spec.ts

```diff
@@ -73,70 +81,216 @@ describe('EditCraComponent', () => {
-    route.queryParams = of({ data: testEditToken });
-    fixture.detectChanges();
-    component.ngOnInit();
-    fixture.whenStable().then(() => {
-      expect(component.mode).toBe(testData.mode);
-      expect(component.cra).toBe(testData.cra);
+        const spySetTitle = spyOn(titleService, 'setTitle');
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(spySetTitle).toHaveBeenCalledWith('Acrabadabra - Saisir un compte rendu d\'activité');
+      });
+
+      it('should init empty', () => {
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(component.consultantNameInput.value).toBe('');
+        expect(component.consultantEmailInput.value).toBe('');
+        expect(component.missionTitleInput.value).toBe('');
+        expect(component.missionFinalClientInput.value).toBe('');
+        expect(component.mode).toBe(testData.mode);
+        expect(component.cra).toEqual(new Cra());
+      });
     });
-  }));
 
-  it('should switch mode when review QueryParams provided', async(() => {
-    testData.mode = 'review';
-    const spy = spyOn(serializer, 'deserialize')
-      .and.callFake(
-        (data) => {
-          expect(data).toBe(testReviewToken);
-        }
-      )
-      .and.returnValue(testData);
-    route.queryParams = of({ data: testReviewToken });
-    fixture.detectChanges();
-    component.ngOnInit();
-    fixture.whenStable().then(() => {
-      expect(component.mode).toBe(testData.mode);
-      expect(component.cra).toBe(testData.cra);
+    describe('when called in edit/review mode', () => {
+      it('should init filled with token data', async(() => {
+        testData.mode = 'edit';
+        route.queryParams = of({ data: testEditToken });
+
+        const spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        fixture.whenStable().then(() => {
+          expect(component.cra).toBe(testData.cra);
+          expect(component.consultantNameInput.value).toBe(testData.cra.consultant.name);
+          expect(component.consultantEmailInput.value).toBe(testData.cra.consultant.email);
+          expect(component.missionTitleInput.value).toBe(testData.cra.mission.title);
+          expect(component.missionFinalClientInput.value).toBe(testData.cra.mission.client);
+          expect(spyDeserialize).toHaveBeenCalledWith(testEditToken);
+        });
+      }));
+    });
+
+    describe('when called in edit mode', () => {
+      it('should set mode to edit', () => {
+        testData.mode = 'edit';
+        route.queryParams = of({ data: testEditToken });
+
+        const spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(component.mode).toBe('edit');
+        expect(spyDeserialize).toHaveBeenCalledWith(testEditToken);
```

Comment:
your test purpose is : ```should set mode to edit```
Know if the `spyDeserialize` was correctly called doesn't interest us, if the token is not the good one, the previous expect with fail (only if you don't mock the return of the `deserialize` method)